src/cli/runtime/http.rs

```diff
@@ -59,6 +60,28 @@ impl RequestCounter {
     }
 }
 
+fn get_memory_by_percentage(percentage: String) -> u64 {
+    // Create a new system object
+    let mut sys = System::new();
+    // Refresh system information
+    sys.refresh_all();
+    // Get total memory in bytes
+    let total_memory: u64 = sys.get_total_memory() * 1024;
+
+    // Parse the percentage string to a float
+    let clamped_percentage: f64 = percentage
+        .parse()
+        .map(|p| if p > 100.0 { 100.0 } else { p })
+        .unwrap_or(0.0);
+
+    // Calculate the memory based on the percentage, without losing precision
+    let memory_by_percentage = (total_memory as f64 * clamped_percentage) as u64;
+
+    // Clamp the memory_by_percentage to u64::MAX
+
+    std::cmp::min(memory_by_percentage, u64::MAX)
+}
```

Comment:
Can move this logic to a separate file `cache_memory.rs`. Should be calculated once and stored statically.

Reply:
done